module/VuFind/src/VuFind/Search/EDS/Options.php

```diff
@@ -154,6 +186,7 @@ public function __construct(
      */
     public function getModeOptions()
     {
+        $this->setOptionsFromApi();
```

Comment:
Yes, any that access API options need to be overridden, and I had missed that one.

Reply:
@demiankatz Weird, I thought I commented about it, but probably lost the plot somehow. While the approach you suggested would be more elegant, it would require us to know the default to return. Since e.g. `$this->defaultLimit` can be null until API options are processed, we don't know the default yet. Or I misunderstood something. :)

Reply:
I don't think we need to know a default -- I just included the second parameter in my example getApiProperty method in case we needed the option to provide one, and I thought it was a good idea to have a fallback in case an unset property was requested. However, in current practice, I wouldn't expect this to get used. It seems like most of the methods have been refactored to: "call setOptionsFromApi, then return a property." So I thought if we made a function that just takes the property name and applies that pattern of behavior, it makes things a little more consistent/readable without changing functionality at all.

Reply:
Now added.